src/MessageHandler/SmsNotificationHandler.php

```diff
@@ -0,0 +1,42 @@
+<?php
+
+declare(strict_types=1);
+
+namespace App\MessageHandler;
+
+use App\Message\SmsNotification;
+use App\Repository\UserRepository;
+use Psr\Log\LoggerAwareInterface;
+use Psr\Log\LoggerAwareTrait;
+use Symfony\Component\Messenger\Handler\MessageHandlerInterface;
+use Symfony\Contracts\HttpClient\HttpClientInterface;
+
+class SmsNotificationHandler implements MessageHandlerInterface, LoggerAwareInterface
+{
+    use LoggerAwareTrait;
+
+    private UserRepository $userRepository;
+
+    private HttpClientInterface $httpClient;
+
+    public function __construct(
+        HttpClientInterface $smsClient,
+        UserRepository $userRepository
+    ) {
+        $this->httpClient = $smsClient;
+        $this->userRepository = $userRepository;
+    }
+
+    public function __invoke(SmsNotification $message): void
+    {
+        $users = $this->userRepository->findAll();
```

Comment:
You call this method before firing the messages and then again here. You need to do it once. Either there for both SMS and emails or here for both SMS and email